### DISCUSSION TITLE
Assistance Required for Mapping Product Table with Custom Table

### DISCUSSION THREAD

**Discussion by Nsr19** [medusajs/medusa#10533] [Q&A]
Hi Team,

I have a requirement where I need to map the existing product table to a new custom table. However, I am unable to locate the model class for the product table in the code base where I can add a column to establish a foreign key relationship.

Could you please assist me with this?

Thank you!

**Comment by olivermrbl** [comment]
Can I get you to explain the use case in more detail?

You cannot extend existing tables in Medusa v2–you will need to create a [link](https://docs.medusajs.com/learn/fundamentals/module-links) between your custom model (with the custom column) and the existing product table.

The product model can be found [here](https://github.com/medusajs/medusa/blob/develop/packages/modules/product/src/models/product.ts).
